lib/entities/scheduled-action.ts

```diff
@@ -33,13 +33,17 @@ type SchedulableActionType = 'publish' | 'unpublish'
 export type ScheduledActionSysProps = {
   id: string
   type: 'ScheduledAction'
+  version?: number
```

Comment:
I am concerned about the optional version here. Isn't version a consistent property that starts with 1 and gets increased with updates?

Reply:
I think for the SDK it can be mandatory (without `?`) -- for some other projects it might break since the version has not been introduced fully yet.

Reply:
It will require typing check, true. But if you leave it optional when in fact it's required, then you'd have to cast things to string or check against undefined each time, which can be even more annoying than updating mocks/fixtures by adding version to it

Reply:
I mean I understand the concern, but should we then mark each new property as optional because it didn't use to be there before? Might not be a great approach

Reply:
Changing
```
createdBy: { sys: MetaSysLink }
```
to
```
createdBy: Link<'User'>
```

might be as breaking ,because what if I set my mock there to be "Person" instead of "User"? :)

Reply:
https://github.com/contentful/contentful-management.js/pull/872#discussion_r651896135

I think you misunderstood, for the SDK I agree with it being mandatory.

Reply:
> Changing
> 
> ```
> createdBy: { sys: MetaSysLink }
> ```
> 
> to
> 
> ```
> createdBy: Link<'User'>
> ```
> 
> might be as breaking ,because what if I set my mock there to have linkType of "Person" instead of "User" and now I updated the sdk and it broke? :)

Why would you set it to `Person`? 🤔 

I don't mind either tbh, `Link<'User'>` is actually what it is but `SysLink` is also a good abstraction.

Reply:
😀 Okay. I'll change it to `SysLink`

Reply:
🤦🏽  -- forgot that we are _adding_  fields here  as well -- for those I'd recommend using the correct `Link<'Type'>` - existing ones can use `SysLink`